explainer-datamodel.md

```diff
@@ -0,0 +1,413 @@
+supported properties like `width` and `height`.
+
+In some cases, we imagine it will be useful to allow the `FormattedText` object to assume
+alternate layout container types where those alternate types provide unique text layout
+capabilities. For example, we expect to support `FormattedText` objects with an inner 
+display type of `ruby` in order to become a Ruby container and enable the use of Ruby
+annotated layout. Other container types are not currently planned to initially support 
+but are good long-term candidates (e.g., multi-column containers created via the `columns`
+shorthand property), while still others are less-likely to be supported (e.g., flex and 
+grid container types, which are less useful for formatted text scenarios).
+
+There are various CSS properties that provide helpful graphical emphasis to text that are 
+also supported. These are for convenience in supporting common text formatting scenarios
+that would otherwise require detailed introspection of the object model's related metrics
+in order to correctly layout and render as desired with respect to the text. Because these
+features are already available in CSS layout engines and significantly ease author burden, 
+many of these CSS properties will be supported. Some supported examples include: 
+`text-decoration`, `text-shadow`, `box-shadow`, even `border`, `outline`, and limited `background` support 
+(where the metrics and composition processing do not require external dependencies, such as 
+image resources typically loaded by `url()` functions).
+
+#### Text clipping and future extensions 
+
+By leveraging CSS, we get the added benefit of a well-known constraint language for 
+expressing box bounds and the expected behavior for content (in our case formatted text 
+content) that overflows those bounds. `width` and `height` and corresponding `min-width`
+or `max-height` properties express the desired constraints for eventual line-box 
+computation when it comes to rendering the object model or returning metrics. Similarly,
+`overflow` and `clip-path` can further ensure the text content expands or is clipped to
+fit the desired constraints. These properties would only apply on the `FormattedText` 
+container object.
+
+CSS continues to evolve, and it makes sense to extend relevant new CSS properties to
+this object model as they become a part of the web platform. For example, while not
+widely supported at the time of writing, support for `shape-inside` (CSS Shapes L2)
+and CSS Exclusions provide exciting growth opportunities for text using this model.
+
+## WebIDL
+
+```webidl
+[Exposed=Window,Worker] 
+interface FormattedText { 
+  constructor();
+  constructor(DOMString textRunText...);
+  constructor(sequence<FormattedTextRunInit> textRunInit);
+  constructor(FormattedText copyConstructorInit);
+
+  attribute ObservableArray<FormattedTextRun> textruns; 
+}; 
+FormattedText includes FormattedTextStylable; 
+
+dictionary FormattedTextRunInit { 
```

Comment:
if we drop font, color perhaps this dictionary is not required. We would have FormattedTextRun constructor that takes text and perhaps in the future a style string.

Reply:
(I put that "in the future a style string" into the spec 🙄. Propose keeping it for now until we get more feedback.)